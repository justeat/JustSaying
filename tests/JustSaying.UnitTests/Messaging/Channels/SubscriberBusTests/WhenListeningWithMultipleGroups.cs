using System.Collections.Generic;
using JustSaying.AwsTools.MessageHandling;
using JustSaying.Messaging.Channels.Configuration;
using JustSaying.Messaging.Channels.SubscriptionGroups;
using Xunit.Abstractions;

namespace JustSaying.UnitTests.Messaging.Channels.SubscriberBusTests
{
    public class WhenListeningWithMultipleGroups : BaseSubscriptionBusTests
    {
        private readonly ISqsQueue _queueB;
        private readonly ISqsQueue _queueA;

        public WhenListeningWithMultipleGroups(ITestOutputHelper testOutputHelper) : base(testOutputHelper)
        {
            _queueA = CreateSuccessfulTestQueue(new TestMessage());
            _queueB = CreateSuccessfulTestQueue(new TestMessage());

        }

        protected override Dictionary<string, SubscriptionGroupSettingsBuilder> SetupBusConfig(SubscriptionConfig config)
        {
            return new Dictionary<string, SubscriptionGroupSettingsBuilder>
            {
                { "queueA", new SubscriptionGroupSettingsBuilder(config).AddQueue(_queueA) },
                { "queueB", new SubscriptionGroupSettingsBuilder(config).AddQueue(_queueB) }
            };
        }

        protected override void Given()
        {
            Queues.Add(_queueA);
            Queues.Add(_queueB);
        }
    }
}
