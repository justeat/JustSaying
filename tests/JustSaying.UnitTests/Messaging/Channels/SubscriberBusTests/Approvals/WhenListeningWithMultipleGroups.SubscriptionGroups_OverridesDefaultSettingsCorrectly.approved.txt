{
  "Groups": [
    {
      "Settings": {
        "ConcurrencyLimit": 1,
        "BufferSize": 20,
        "MultiplexerCapacity": 30,
        "Prefetch": 5,
        "Name": "queueA",
        "Queues": [
          {
            "QueueName": "",
            "RegionSystemName": "",
            "Uri": "http://foo.com"
          }
        ]
      }
    },
    {
      "Settings": {
        "ConcurrencyLimit": 48,
        "BufferSize": 10,
        "MultiplexerCapacity": 100,
        "Prefetch": 10,
        "Name": "queueB",
        "Queues": [
          {
            "QueueName": "",
            "RegionSystemName": "",
            "Uri": "http://foo.com"
          }
        ]
      }
    }
  ]
}